tests/kafkatest/services/kafka/kafka.py

```diff
@@ -56,24 +72,34 @@ def start(self):
                 self.create_topic(topic_cfg)
 
     def start_node(self, node):
-        props_file = self.render('kafka.properties', node=node, broker_id=self.idx(node))
+        cfg = KafkaConfig(**node.config)
+        cfg[kafka_prop.ADVERTISED_HOSTNAME] = node.account.hostname
+        cfg[kafka_prop.ZOOKEEPER_CONNECT] = self.zk.connect_setting()
+        props_file = cfg.render()
         self.logger.info("kafka.properties:")
         self.logger.info(props_file)
         node.account.create_file("/mnt/kafka.properties", props_file)
 
-        cmd = "/opt/kafka/bin/kafka-server-start.sh /mnt/kafka.properties 1>> /mnt/kafka.log 2>> /mnt/kafka.log & echo $! > /mnt/kafka.pid"
+        cmd = "export LOG_DIR=/mnt/kafka-operational-logs/; "
+        cmd += self._kafka_dir(node) + "/bin/kafka-server-start.sh /mnt/kafka.properties 1>> /mnt/kafka.log 2>> /mnt/kafka.log &"
         self.logger.debug("Attempting to start KafkaService on %s with command: %s" % (str(node.account), cmd))
         with node.account.monitor_log("/mnt/kafka.log") as monitor:
             node.account.ssh(cmd)
             monitor.wait_until("Kafka Server.*started", timeout_sec=30, err_msg="Kafka server didn't finish startup")
-        if len(self.pids(node)) == 0:
-            raise Exception("No process ids recorded on node %s" % str(node))
+
+    def _kafka_dir(self, node):
+        if node.version == KafkaVersion.TRUNK:
+            return "/opt/kafka"
+        else:
+            return "/opt/kafka-" + node.version
 
     def pids(self, node):
         """Return process ids associated with running processes on the given node."""
         try:
-            return [pid for pid in node.account.ssh_capture("cat /mnt/kafka.pid", callback=int)]
-        except:
+            cmd = "ps ax | grep -i kafka.properties | grep java | grep -v grep | awk '{print $1}'"
```

Comment:
@ewencp When/how does this break?  Is it because some systems limit the output length of `ps`, and therefore the search string might not be present? 

Several existing services already use this approach. Furthermore, grepping the output of `ps` to find and kill process ids is what underlies every single cleanup method, so if this approach truly brittle...

If the only problem is command length, it would be fairly simple to shorten it to something reliably greppable: copy the given too-long command to a remote temporary script with a greppable name, and run the script.